### DISCUSSION TITLE
Unable to compile (MEF) project after version 88.2.90 due to not finding config file

### DISCUSSION THREAD

**Discussion by bgoldstein14** [cefsharp/CefSharp#3611] [General]
My apologies ahead of time, but this will seem a bit general as I'm not entirely sure what's going on, however, we can definitely see an issue between versions in our MEF project and I'm not 100% certain it's MEF related but it seems like it's a good chance.

Our project uses a config file that's the same name as the project which is required as part of the MEF build. The build error is shown below. The file is most definitely in the directory in question.

`Could not open Source file: Could not find a part of the path 'C:\Users\B12345\source\repos\eQ\Janus.Portal.Plugins.Equity.Coverage\bin\Debug\Janus.Portal.Plugins.Equity.Coverage.dll.config;bin\Debug\Janus.Portal.Plugins.Equity.Coverage.dll.config'.`

I've tested upgrading to 90.6.70 and get the above failure. Downgrading to 88.2.90 removes the failure. I've experimented with versions in between to get the same failure. Unfortunately, our project/solution is quite massive and I don't have a condensed reproducible scenario to upload. I'll understand if this gets low/no priority due to that. But, at some point our company will need to deal with this if we ever want to upgrade versions and perhaps we'll try to come up with some sort of new solution that reproduces this for eventual upload.

Is there a workaround or additional setting I'm missing with the upgraded versions? My apologies if I missed this somewhere in the documentation.

**Comment by amaitland** [comment]
> The file is most definitely in the directory in question.

The path in the example provided would be invalid, is that just a quirk of the exception message? 

> I've tested upgrading to 90.6.70 and get the above failure. Downgrading to 88.2.90 removes the failure. 

How does version 89 behave? Be helpful to narrow the version down.

> Is there a workaround or additional setting I'm missing with the upgraded versions? 

I'm not aware of anything thatwould cause this, unfortunately there's not enough information to speculate what's going on.

Every new version of Chromium brings the potential for changes in behaviour,  in terms of CefSharp changes you can compare the two release branches and see if anything stands out.

https://github.com/cefsharp/CefSharp/compare/cefsharp/88...cefsharp/90

**Reply by bgoldstein14** [reply]
The path is correct. What you're seeing is the entire copy of the message from Visual Studio and it duplicates the name of the "missing" file after the full path.

At some point I can try 89, but for now we have to concentrate on other issues with deploying the component.

I don't know how/why a different version of Chromium would harm the build process. If we get rid of the config file from the project (which we really can't do, but do it for testing) then it builds fine. Again, I have to wonder if MEF isn't playing nice for some reason here.

**Reply by amaitland** [reply]
> I don't know how/why a different version of Chromium would harm the build process. 

It's just a general statement, it's important that users realize that many of the differences they see when upgrading aren't direct changes to CefSharp, they are upstream changes. Version 90 for example includes a different set of pak files as a result of CEF changes https://github.com/cefsharp/CefSharp/issues/3509#issuecomment-821756091

Whenever you see a change in behaviour between versions it's important to stop and think where the change might originate, CefSharp, CEF, Chromium. Again,  this is a general statement. 

As you are seeing an error from msbuild it's likely chromium changes are related in this particular instance.
